### PR TITLE
Adds details about required kernel versions above 4.9.17, supported OS update

### LINKED ISSUE #10534: Docs: Update System Requirements to call out key features/functionality and required kernel versions
## Bug report

Presently, the Cilium documentation has information buried deep within it that call out specific kernel versions required for various cilium features. In order to assist users with finding this information earlier in their discovery process, it should be moved to the System Requirements page to be maintained as new features are added which are kernel dependent.

**General Information**
kube-proxy free and 4.19.57 requirement - https://cilium.readthedocs.io/en/stable/gettingstarted/kubeproxy-free/

host reachable services and 4.19.57 requirement - https://docs.cilium.io/en/v1.7/gettingstarted/host-services/

Sockmaps - 4.19.?? - not documented - only hit = https://docs.cilium.io/en/v1.7/architecture/#datapath
kTLS - 4.19.?? - not documented - only hit = https://docs.cilium.io/en/v1.7/envoy/extensions/#envoy-go-extensions

**How to reproduce the issue**

1. Search the docs for above terms

### PR DESCRIPTION
**Pull request by seanmwinn**
Adds details to System Requirements about cilium features and required kernel versions above 4.9.17, supported OS update
Fixes: #10534
Signed-off-by: Sean Winn <sean@isovalent.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10537)

### CODE REVIEW

File: Documentation/install/system_requirements.rst [hunk 1]
```diff
@@ -54,24 +54,35 @@ Linux Distribution Compatibility Matrix
...
-`Fedora Atomic/Core`_ >= 25
-LinuxKit_             all
-Ubuntu_               >= 16.04.2, >= 16.10
-Opensuse_             Tumbleweed, >=Leap 15.0
-===================== ====================
-
+========================== ====================
+Distribution               Minimum Version
+========================== ====================
+`Amazon Linux 2`_          >= 4.15
```
Comment (joestringer) [inline Documentation/install/system_requirements.rst]: Is this a version of the distribution or the kernel that it runs on?

File: Documentation/install/system_requirements.rst [hunk 2]
```diff
@@ -136,6 +147,26 @@ by adding the following to the helm configuration command line:
...
+kernel features and functionality continues to progress in the Linux community.
+Some Cilium features and functionality are dependent on newer kernel versions.
+These additional Cilium features and functionality are enabled by upgrading to
+a later kernel version as detailed below:
+
+========================== ==========================
+Cilium Feature             Minimum Kernel Version
+========================== ==========================
+`Longest Prefix Match`_    4.11+
+`Host Reachable Services`_ 4.19.57+
```
Comment (joestringer) [inline Documentation/install/system_requirements.rst]: Might be worth adding the other (v5.1.16+, v5.2.0+) kernel versions here too, per the `host-services` doc callout?

Comment (joestringer) [reply Documentation/install/system_requirements.rst]: I imagine that this the same for kube-proxy free as well.

**Review by joestringer — CHANGES_REQUESTED** [review]
I also see references to `4.9.17` in `Documentation/concepts/overview.rst`, though maybe that document should just point over to here for the kernel requirements now instead of having its own version content / text?

While you're at it, you may consider adding RancherOS per https://github.com/cilium/cilium/issues/9825 as well.

Couple of minor fixup requests below.

File: Documentation/install/system_requirements.rst (hunk 1 above)
Comment (seanmwinn) [reply Documentation/install/system_requirements.rst]: Updated amazon Linux 2 to reflect support for all versions

File: Documentation/install/system_requirements.rst (hunk 2 above)
Comment (seanmwinn) [reply Documentation/install/system_requirements.rst]: fixed up formatting for the kernel versions, added recommendations

File: Documentation/install/system_requirements.rst [hunk 3]
```diff
@@ -136,6 +147,26 @@ by adding the following to the helm configuration command line:
...
+
+========================== ===============================
+Cilium Feature             Minimum Kernel Version
+========================== ===============================
+`Longest Prefix Match`_    >= 4.11
+`Host Reachable Services`_ >= 4.19.57, >= 5.1.16+,  >=5.2+
+`Kube-proxy Free`_         >= 4.19.57, >= 5.1.16+,  >=5.2+
+========================== ===============================
+
+.. _Longest Prefix Match: |SCM_WEB|\/install/upgrade/#restrictions-on-unique-prefix-lengths-for-cidr-policy-rules
```
Comment (qmonnet) [inline Documentation/install/system_requirements.rst]: When available we also use LRU (Least Recently Used) hash maps with BPF, which appeared [in kernel 4.10](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=29ba732acbeece1e34c68483d1ec1f3720fa1bb3). But as far as I know we don't have any specific documentation on the topic (not sure it makes a difference to the user, other than for performance), so not sure if worth adding in this table.

Comment (joestringer) [reply Documentation/install/system_requirements.rst]: I think for current usage (existing maps) we work around the lack of LRU by just creating a hashmap and spinning up a garbage collection controller in the cilium-agent.

**Comment by joestringer** [comment]
test-docs-please

**Review by joestringer — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

File: Documentation/install/system_requirements.rst [hunk 4]
```diff
@@ -136,6 +147,26 @@ by adding the following to the helm configuration command line:
      ...
      --set global.enableXTSocketFallback=false
 
+Advanced Features and Required Kernel Version
+=============================================
+Cilium requires Linux Kernel 4.9.17 or higher, however development on additional
```
Comment (pchaigno) [inline Documentation/install/system_requirements.rst]: Nit: `Linux kernel` has a lowercase `k` elsewhere in the documentation.

File: Documentation/install/system_requirements.rst [hunk 5]
```diff
@@ -136,6 +147,26 @@ by adding the following to the helm configuration command line:
...
+Cilium Feature             Minimum Kernel Version
+========================== ===============================
+`Longest Prefix Match`_    >= 4.11
+`Host Reachable Services`_ >= 4.19.57, >= 5.1.16+,  >=5.2+
+`Kube-proxy Free`_         >= 4.19.57, >= 5.1.16+,  >=5.2+
+========================== ===============================
+
+.. _Longest Prefix Match: |SCM_WEB|\/install/upgrade/#restrictions-on-unique-prefix-lengths-for-cidr-policy-rules
+.. _Host Reachable Services: |SCM_WEB|\/gettingstarted/host-services/
+.. _Kube-proxy Free: |SCM_WEB|\/gettingstarted/kubeproxy-free
```
Comment (pchaigno) [inline Documentation/install/system_requirements.rst]: These three links seem broken. I think they should use the `:ref:` syntax instead, as seen in the e.g. `Key-Value store` section of this same document.

EDIT: You can use `make render-docs` locally to previous the documentation.

**Review by pchaigno — CHANGES_REQUESTED** [review]

File: Documentation/install/system_requirements.rst (hunk 5 above)
Comment (seanmwinn) [reply Documentation/install/system_requirements.rst]: fixed, I have been using make render-docs but missed these broken links, thanks

File: Documentation/install/system_requirements.rst (hunk 4 above)
Comment (seanmwinn) [reply Documentation/install/system_requirements.rst]: lowercase it is

**Review by pchaigno — APPROVED** [review]

File: Documentation/install/system_requirements.rst [hunk 6]
```diff
@@ -136,6 +147,22 @@ by adding the following to the helm configuration command line:
...
+Some Cilium features and functionality are dependent on newer kernel versions.
+These additional Cilium features and functionality are enabled by upgrading to
+a later kernel version as detailed below:
+
+======================= ===============================
+Cilium Feature          Minimum Kernel Version
+======================= ===============================
+:ref:`cidr_limitations` >= 4.11
+:ref:`host-services`    >= 4.19.57, >= 5.1.16+,  >=5.2+
+:ref:`kubeproxy-free`   >= 4.19.57, >= 5.1.16+,  >=5.2+
```
Comment (joestringer) [inline Documentation/install/system_requirements.rst]: Just noticed this is a bit inconsistent, the `>=` already implies the `+`:
```suggestion
:ref:`host-services`    >= 4.19.57, >= 5.1.16,  >= 5.2
:ref:`kubeproxy-free`   >= 4.19.57, >= 5.1.16,  >= 5.2
```

**Review by joestringer — CHANGES_REQUESTED** [review]
One more minor nit.

**Comment by tklauser** [comment]
@seanmwinn @joestringer what's the status of this? I see one small open suggestion which hasn't been incorporated yet. Since this PR is tagged as `needs-backport-1.6` and `needs-backport-1.7` it would be great to get it merged to `master` soon, such that we can pick it up for the next round of backports. Thanks.

**Comment by joestringer** [comment]
I'll just fold in the suggestion since it's minor and push it through with a 'Squash and Merge'

**Comment by joestringer** [comment]
test-docs-please

**Pull request merged**